processing/src/main/java/org/apache/carbondata/processing/loading/jsoninput/JsonInputFormat.java

```diff
@@ -0,0 +1,285 @@
+
+    private Logger LOG = Logger.getLogger(JsonRecordReader.class);
+
+    private JsonStreamReader rdr = null;
+
+    private long start = 0, end = 0;
+
+    private float toRead = 0;
+
+    private String identifier = null;
+
+    private Logger log = Logger.getLogger(JsonRecordReader.class);
+
+    private Text outJson = new Text();
+
+    private LongWritable outKey = new LongWritable();
+
+    @Override public void initialize(InputSplit split, TaskAttemptContext context)
+        throws IOException, InterruptedException {
+
+      this.identifier = context.getConfiguration().get(RECORD_IDENTIFIER);
+
+      if (this.identifier == null || identifier.isEmpty()) {
+        throw new InvalidParameterException(JsonInputFormat.RECORD_IDENTIFIER + " is not set.");
+      } else {
+        LOG.info("Initializing JsonRecordReader with identifier " + identifier);
+      }
+
+      FileSplit fSplit = (FileSplit) split;
+
+      // get relevant data
+      Path file = fSplit.getPath();
+
+      log.info("File is " + file);
+
+      start = fSplit.getStart();
+      end = start + split.getLength();
+      toRead = end - start;
+
+      FSDataInputStream strm = FileSystem.get(context.getConfiguration()).open(file);
+
+      if (start != 0) {
+        strm.seek(start);
+      }
+
+      rdr = new JsonStreamReader(identifier, new BufferedInputStream(strm));
+    }
+
+    @Override public boolean nextKeyValue() throws IOException, InterruptedException {
+      boolean retval = false;
+      boolean keepGoing = false;
+      do {
+        keepGoing = false;
+        String record = rdr.getJsonRecord();
+        if (record != null) {
+          if (JsonInputFormat.decodeLineToJsonNode(record) == null) {
+            keepGoing = true;
+          } else {
+            outJson.set(record);
+            outKey.set(rdr.getBytesRead());
+            retval = true;
+          }
+        }
+      } while (keepGoing);
+
+      return retval;
+    }
+
+    @Override public void close() throws IOException {
+      rdr.close();
+    }
+
+    @Override public float getProgress() throws IOException {
+      return (float) rdr.getBytesRead() / toRead;
+    }
+
+    @Override public LongWritable getCurrentKey() throws IOException, InterruptedException {
+      return outKey;
+    }
+
+    @Override public Text getCurrentValue() throws IOException, InterruptedException {
+      return outJson;
+    }
+  }
+
+  /**
+   * Decodes a given string of text to a {@link JsonNode}.
+   *
+   * @param line The line of text
+   * @return The JsonNode or null if a JsonParseException,
+   * JsonMappingException, or IOException error occurs
+   */
+  public static synchronized JsonNode decodeLineToJsonNode(String line) {
+
+    try {
+      return mapper.readTree(line);
+    } catch (JsonParseException e) {
+      e.printStackTrace();
+      return null;
+    } catch (JsonMappingException e) {
+      e.printStackTrace();
```

Comment:
remove stack trace and add logger and throw the exception.

Reply:
ok. done